OpenRA.Game/FileFormats/ReplayReport.cs

```diff
@@ -0,0 +1,290 @@
+								text.AppendLine("\tSpawn point: " + player.SpawnPoint);
+								text.AppendLine("\tRandom spawn point: " + player.IsRandomSpawnPoint);
+
+								if (player.IsHuman && !gameInfo.IsSinglePlayer)
+									foreach (var line in NetworkInfo(session, client))
+										text.AppendLine("\t" + line);
+
+								text.AppendLine();
+							}
+						}
+
+						var spectators = session.Clients.Where(c => c.IsObserver);
+						if (spectators.Any())
+						{
+							var specLabel = "Spectators:";
+							text.AppendLine(specLabel);
+							text.AppendLine(new string('-', specLabel.Length));
+							foreach (var spec in spectators)
+							{
+								text.AppendLine(spec.Name);
+								if (!gameInfo.IsSinglePlayer)
+								{
+									if (spec.IsAdmin)
+										text.AppendLine("\t[Admin]");
+
+									foreach (var line in NetworkInfo(session, spec))
+										text.AppendLine("\t" + line);
+								}
+
+								text.AppendLine();
+							}
+						}
+
+						foreach (var line in SettingsDisplay(session.GlobalSettings))
+							text.AppendLine(line);
+
+						text.AppendLine();
+						gameStarted = true;
+					}
+				},
+				{ "Chat", s =>
+					{
+						if (gameStarted)
+							text.Append(time);
+
+						var client = session.ClientWithIndex(packetClientIndex);
+						text.Append(client.Name);
+						text.Append(client.IsObserver ? " (Spectator): " : ": ");
+						text.AppendLine(s);
+					}
+				},
+				{ "TeamChat", s =>
+					{
+						if (gameStarted)
+							text.Append(time);
+
+						text.AppendLine(session.ClientWithIndex(packetClientIndex).Name + " (Team): " + s);
+					}
+				},
+				{ "Message", s =>
+					{
+						foreach (var line in s.Split('\n'))
+						{
+							if (gameStarted)
+								text.Append(time);
+
+							text.AppendLine("Server: " + line);
+						}
+					}
+				}
+			};
+
+			try
+			{
+				using (var fs = File.OpenRead(metadata.FilePath))
+				{
+					var networkTickCount = 0;
+
+					while (fs.Position < metadata.MetaStartMarkerPosition)
+					{
+						packetClientIndex = fs.ReadInt32();
+						var packetLen = fs.ReadInt32();
+						var packet = fs.ReadBytes(packetLen);
+
+						if (packet.Length == 5 && packet[4] == 0xBF)
+							continue; // disconnect
+
+						if (packet.Length >= 5 && packet[4] == 0x65)
+							continue; // sync
+
+						var frame = BitConverter.ToInt32(packet, 0);
+						networkTickCount = Math.Max(networkTickCount, frame);
+						var orders = packet.ToOrderList(null);
+
+						// Time stamps are in game time.
+						time = new TimeSpan(0, 0, networkTickCount * networkTickDuration / 1000) + " ";
+
+						foreach (var o in orders)
+						{
+							if (!orderHandler.ContainsKey(o.OrderString))
+								continue;
+
+							orderHandler[o.OrderString](o.TargetString);
+						}
+					}
+				}
+			}
+			catch (Exception ex)
+			{
+				// OutOfMemoryException can be thrown if the replay file is corrupt.
+				if (ex is ArgumentException || ex is NotSupportedException || ex is IOException
```

Comment:
Catching Exception and then rethrowing it is also done elsewhere in the game. Here for example:
https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Utility/Program.cs#L56